cmd/minishift/cmd/config/config.go

```diff
@@ -38,84 +37,66 @@ type setFn func(string, string) error
-		name: "hostfolders-automount",
-		set:  SetBool,
-	},
-	{
-		name: "hostfolders-mountpath",
-		set:  SetString,
-	},
-	{
-		name: config.WantUpdateNotification,
-		set:  SetBool,
-	},
-	{
-		name: config.ReminderWaitPeriodInHours,
-		set:  SetInt,
-	},
+var flagsList []Setting
+
+var (
+	// minishift
+	ISOUrl           = newFlags("iso-url", SetString, []setFn{IsValidUrl}, []setFn{RequiresRestartMsg}, true)
+	CPUs             = newFlags("cpus", SetInt, []setFn{IsPositive}, []setFn{RequiresRestartMsg}, true)
+	Memory           = newFlags("memory", SetInt, []setFn{IsPositive}, []setFn{RequiresRestartMsg}, true)
+	DiskSize         = newFlags("disk-size", SetString, []setFn{IsValidDiskSize}, []setFn{RequiresRestartMsg}, true)
+	VmDriver         = newFlags("vm-driver", SetString, []setFn{IsValidDriver}, []setFn{RequiresRestartMsg}, true)
+	OpenshiftVersion = newFlags("openshift-version", SetString, nil, nil, true)
+	HostOnlyCIDR     = newFlags("host-only-cidr", SetString, []setFn{IsValidCIDR}, nil, true)
+
+	// cluster up
+	SkipRegistryCheck = newFlags("skip-registry-check", SetBool, nil, nil, true)
+	PublicHostname    = newFlags("public-hostname", SetString, nil, nil, true)
+	RoutingSuffix     = newFlags("routing-suffix", SetString, nil, nil, true)
+	HostConfigDir     = newFlags("host-config-dir", SetString, []setFn{IsValidPath}, nil, true)
+	HostVolumeDir     = newFlags("host-volumes-dir", SetString, []setFn{IsValidPath}, nil, true)
+	HostDataDir       = newFlags("host-data-dir", SetString, []setFn{IsValidPath}, nil, true)
+	HostPvDir         = newFlags("host-pv-dir", SetString, []setFn{IsValidPath}, nil, true)
+	ServerLogLevel    = newFlags("server-loglevel", SetInt, []setFn{IsPositive}, nil, true)
+	OpenshiftEnv      = newFlags("openshift-env", nil, nil, nil, false)
+	Metrics           = newFlags("metrics", SetBool, nil, nil, true)
+	Logging           = newFlags("logging", SetBool, nil, nil, true)
+
+	// Setting proxy
+	NoProxyList = newFlags("no-proxy", SetString, nil, nil, true)
+	HttpProxy   = newFlags("http-proxy", SetString, []setFn{IsValidProxy}, nil, true)
+	HttpsProxy  = newFlags("https-proxy", SetString, []setFn{IsValidProxy}, nil, true)
+
+	// Subscription Manager
+	Username = newFlags("username", SetString, nil, nil, true)
+	Password = newFlags("password", SetString, nil, nil, true)
+
+	// Global flags
+	LogDir             = newFlags("log_dir", SetString, []setFn{IsValidPath}, nil, true)
+	ShowLibmachineLogs = newFlags("show-libmachine-logs", SetBool, nil, nil, true)
+
+	// Host Folders
+	HostFoldersMountPath = newFlags("hostfolders-mountpath", SetString, nil, nil, true)
+	HostFoldersAutoMount = newFlags("hostfolders-automount", SetBool, nil, nil, true)
+)
+
+func newFlags(name string, set func(MinishiftConfig, string, string) error, validations []setFn, callbacks []setFn, isApply bool) *Setting {
+	flag := Setting{
+		Name:        name,
+		set:         set,
+		validations: validations,
+		callbacks:   callbacks,
+	}
+	if isApply {
```

Comment:
+1 right, a flag is enough. No need to pass the array itself. It would maybe be a bit more explicit, but ok